lib/web_catalog/object_graph_importer.es6.js

```diff
@@ -45,38 +54,116 @@ foam.CLASS({
       required: true,
       final: true,
     },
+    {
+      name: 'numWorkers',
+      value: 16,
+    },
+    {
+      name: 'stubFactory',
+      factory: function() {
+        return this.StubFactorySingleton.create();
+      },
+    },
+    {
+      class: 'FObjectProperty',
+      of: 'org.chromium.apis.web.ApiExtractorService',
+      name: 'apiExtractorService',
+      factory: function() {
+        return this.stubFactory.get(this.ApiExtractorService).create({
+          delegate: this.apiContext_.registry.register(
+            null, null, this.SkeletonBox.create({
+              data: this.ApiExtractorService.create(),
+            })),
+        }, this.containerContext_);
+      },
+    },
+    {
+      class: 'FObjectProperty',
+      of: 'foam.box.Context',
+      name: 'containerContext_',
+      factory: function() {
+        return this.Context.create({
+          unsafe: false,
+          classWhitelist: require('../../data/class_whitelist.json'),
+        });
+      },
+    },
+    {
+      class: 'FObjectProperty',
+      of: 'foam.box.Context',
+      name: 'apiContext_',
+      factory: function() {
+        let ctx = this.Context.create({
+        unsafe: false,
+        classWhitelist: require('../../data/class_whitelist.json'),
+      });
+        ctx.registry = this.BroadcastRegistry.create(null, ctx);
+        const numWorkers = this.numWorkers;
+        let workers = new Array(numWorkers);
+
+        for (let i = 0; i < numWorkers; i++) {
+          // Stub BoxRegistry interface over box addressed to new fork.
+          workers[i] = this.stubFactory.get(this.BoxRegistry).create({
+            delegate: this.ForkBox.create({
+              critical: true,
+              nodeParams:
+              // ['--inspect', '--debug-brk'],
+              this.getForkNodeParams_(),
+              childScriptPath: this.getForkScriptPath_(),
+            }, this.containerContext_),
+          }, this.containerContext_);
+        }
+        ctx.registry.delegates = workers;
+
+        return ctx;
+      },
+    },
   ],
+
   methods: [
     {
       name: 'import',
       documentation: `Reads object graph files from objectGraphPath
           and extract web interfaces and import it to cloudstoreDAO
           using apiImporter.`,
       code: function() {
+        const ogFileRegExp = /window_[a-zA-Z0-9_.]+_[a-zA-Z0-9_.]+_[a-zA-Z0-9_.]+_[a-zA-Z0-9_.]+[.]json$/;
```

Comment:
This makes me think that the collection of filenames could be its own method, or maybe an input to this object. The filename schema has to be right for the parsing to work, which is a bit ugly, but I think that can be addressed through documentation.

Reply:
I've done my best to document this where `objectGraphPath` is defined. I got halfway through refactoring this into accepting an array of paths before realizing that the file name format is being used to extract information, too. I think if this importer gets refactored in the future, that would be a good time to consider other ways to extract the information and make this interface more robust.